### PR TITLE
Feature: add support for DNS filtering

### PR DESCRIPTION
**Pull request by vmanilo**
Resolves https://github.com/Twingate/terraform-provider-twingate/issues/466

Tests passes: https://github.com/vmanilo/terraform-provider-twingate/actions/runs/10374942101/job/28727414984

## Changes
- added resource and datasource for DNS profile

### CODE REVIEW

File: docs/resources/dns_filtering_profile.md [hunk 1]
```diff
@@ -0,0 +1,151 @@
+---
```
Comment (twingate-blee) [inline docs/resources/dns_filtering_profile.md:1]: I think we need to add some text there on how this is a business or enterprise feature and it needs to be enabled before you can use it

File: docs/resources/dns_filtering_profile.md [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+# generated by https://github.com/hashicorp/terraform-plugin-docs
+page_title: "twingate_dns_filtering_profile Resource - terraform-provider-twingate"
+subcategory: ""
+description: |-
+  DNS filtering gives you the ability to control what websites your users can access. For more information, see Twingate's documentation https://www.twingate.com/docs/dns-filtering.
+---
+
+# twingate_dns_filtering_profile (Resource)
+
+DNS filtering gives you the ability to control what websites your users can access. For more information, see Twingate's [documentation](https://www.twingate.com/docs/dns-filtering).
```
Comment (alexb-twingate) [inline docs/resources/dns_filtering_profile.md:11]: Prior to "For more information...", I would add the sentence "DNS filtering is only available on certain plans."

File: docs/data-sources/dns_filtering_profile.md [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+# generated by https://github.com/hashicorp/terraform-plugin-docs
+page_title: "twingate_dns_filtering_profile Data Source - terraform-provider-twingate"
+subcategory: ""
+description: |-
+  DNS filtering gives you the ability to control what websites your users can access. For more information, see Twingate's documentation https://www.twingate.com/docs/dns-filtering.
+---
+
+# twingate_dns_filtering_profile (Data Source)
+
+DNS filtering gives you the ability to control what websites your users can access. For more information, see Twingate's [documentation](https://www.twingate.com/docs/dns-filtering).
```
Comment (alexb-twingate) [inline docs/data-sources/dns_filtering_profile.md:11]: Prior to "For more information...", I would add the sentence "DNS filtering is only available on certain plans."

**Review by alexb-twingate — COMMENTED** [review]
Left some comments about docs. They generally look good, but I would add a callout about tiers, like @bertekintw said.

**Review by twingate-blee — APPROVED** [review]
lgtm

File: docs/data-sources/dns_filtering_profile.md [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+# generated by https://github.com/hashicorp/terraform-plugin-docs
+page_title: "twingate_dns_filtering_profile Data Source - terraform-provider-twingate"
+subcategory: ""
+description: |-
+  DNS filtering gives you the ability to control what websites your users can access. DNS filtering is only available on certain plans. For more information, see Twingate's documentation https://www.twingate.com/docs/dns-filtering.
+---
+
+# twingate_dns_filtering_profile (Data Source)
+
+DNS filtering gives you the ability to control what websites your users can access. DNS filtering is only available on certain plans. For more information, see Twingate's [documentation](https://www.twingate.com/docs/dns-filtering).
```
Comment (alexb-twingate) [inline docs/data-sources/dns_filtering_profile.md:11]: Add line:

> DNS filtering must be enabled for this data source to work. If DNS filtering isn't enabled, the provider will throw an error.

File: docs/resources/dns_filtering_profile.md [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+# generated by https://github.com/hashicorp/terraform-plugin-docs
+page_title: "twingate_dns_filtering_profile Resource - terraform-provider-twingate"
+subcategory: ""
+description: |-
+  DNS filtering gives you the ability to control what websites your users can access. DNS filtering is only available on certain plans. For more information, see Twingate's documentation https://www.twingate.com/docs/dns-filtering.
+---
+
+# twingate_dns_filtering_profile (Resource)
+
+DNS filtering gives you the ability to control what websites your users can access. DNS filtering is only available on certain plans. For more information, see Twingate's [documentation](https://www.twingate.com/docs/dns-filtering).
```
Comment (alexb-twingate) [inline docs/resources/dns_filtering_profile.md:11]: Add line:

> DNS filtering must be enabled for this resources to work. If DNS filtering isn't enabled, the provider will throw an error.

**Pull request merged**